src/services/serverQueryFetch.ts

```diff
@@ -0,0 +1,13 @@
+import * as i from 'types';
+import { dehydrate, QueryClient, QueryKey } from 'react-query';
+
+export const serverQueryFetch = async <Fn extends i.AnyFn>(key: QueryKey, fetchFn: Fn) => {
+  // Create a new QueryClient instance for each page request.
+  // This ensures that data is not shared between users and requests.
+  const queryClient = new QueryClient();
+  await queryClient.prefetchQuery(key, fetchFn);
+
+  return {
+    dehydratedState: dehydrate(queryClient),
```

Comment:
Waarom deze niet destructen zoals het eerst was? Snap dat deze in `_app.tsx` dan niet meer klopt, maar die kunnen we ook aanpassen toch? 

```
type Props = Omit<AppProps, 'pageProps'> & {
  pageProps: i.AnyObject & DehydratedState;
};
```

Zoiets?

Reply:
Anders kan je `dehydratedState` value niet makkelijk doorgeven aan de `<Hydrate />` component in `_app.tsx`

Dan zou je zoiets moeten doen:

```jsx
<Hydrate state={{
  mutations: pageProps.mutations,
  queries: pageProps.queries,
}}>
```
Dat scaled niet zo lekker als de shape verandert.